lua/lvim/core/autocmds.lua

```diff
@@ -34,6 +34,59 @@ function M.load_defaults()
         end,
       },
     },
+    {
+      "BufReadPre",
+      {
+        group = "_general_settings",
+        pattern = "*",
+        desc = "Performance improvements for big files",
+        callback = function(args)
+          if vim.b.big_file then
+            return
+          end
+          local MB = 1024 * 1024
+          local ok, stats = pcall(vim.loop.fs_stat, vim.api.nvim_buf_get_name(args.buf))
+          if not (ok and stats and stats.size > MB) then
+            return
+          end
+          vim.b.big_file = true
+          vim.g.big_files = vim.g.big_files and vim.g.big_files + 1 or 1
+
+          if lvim.builtin.illuminate.active then
+            pcall(require("illuminate.engine").stop_buf, args.buf)
+          end
+
+          local function callback()
+            vim.cmd "NoMatchParen"
+            vim.cmd "syntax off"
+            vim.cmd "syntax clear"
+            vim.cmd "setlocal nolist noswapfile noundofile"
+
+            vim.api.nvim_create_autocmd({ "BufDelete" }, {
+              callback = function()
+                vim.g.big_files = vim.g.big_files - 1
+                if vim.g.big_files == 0 then
+                  vim.cmd "DoMatchParen"
+                  vim.cmd "syntax on"
+                end
+              end,
+              buffer = args.buf,
+            })
+          end
+
+          if stats.size > 5 * MB then
+            callback()
+            return
+          end
+
+          vim.schedule(function()
+            if vim.tbl_contains({ "json" }, vim.bo.ft) then
+              vim.api.nvim_buf_call(args.buf, callback)
+            end
+          end)
```

Comment:
- what does this do now?
- I don't think we need to care about whether it's `json` or not anymore, since the criteria is now only concerned with the file-size

Reply:
json required turning off features with a smaller file size than other languages, but I guess I can just turn off everything on 1MB, it would simplify the rest as well